stubs/networkx/networkx/convert.pyi

```diff
@@ -0,0 +1,16 @@
+from collections.abc import Iterable
+from typing import Any
+from typing_extensions import TypeAlias
+
+import numpy
+import scipy
+from networkx.classes.graph import EdgePlus, Graph, _Node
+
+Data: TypeAlias = (
+    Graph[_Node]
+    | dict[_Node, dict[_Node, dict[str, Any]]]
+    | dict[_Node, Iterable[_Node]]
+    | Iterable[EdgePlus[_Node]]
+    | numpy.ndarray[_Node, Any]
+    | scipy.sparse.base.spmatrix
```

Comment:
I can't find `Data` in the source, is it meant as a type-stub only alias? If so, you can prepend it with `_` to mark it as private, to reduce chances of a user trying to use this alias at runtime.

I can also see that it is only used in one other module (`networkx/classes/graph.pyi`), consider whether the `convert` module is the right place to define this re-usable alias, or maybe it might be better to put it in `networkx/__init__.pyi`.

You're gonna have issues using `scipy` here because it is not yet marked as `py.typed` (see https://github.com/scipy/scipy/issues/17158), so stubsuploader won't allow it in the metadata `requires` field, and `stubtest_requirements` only affects stubtest, not other tooling (like mypy, pyright, pytype, ...).

Optionally, instead of just commenting it out, you could consider creating a Protocol that imitates the shape of the `spmatrix` class, so that if you use `_Data` as a parameter type, `spmatrix` would be a valid type.

```suggestion
from networkx.classes.graph import EdgePlus, Graph, _Node

_Data: TypeAlias = (
    Graph[_Node]
    | dict[_Node, dict[_Node, dict[str, Any]]]
    | dict[_Node, Iterable[_Node]]
    | Iterable[EdgePlus[_Node]]
    | numpy.ndarray[_Node, Any]
    # | scipy.sparse.base.spmatrix  # scipy is not typed yet
```